src/Simulator.cpp

```diff
@@ -458,17 +576,121 @@ std::pair<dd::ComplexValue, std::string> Simulator::getPathOfLeastResistance() {
 
 double Simulator::assign_probs(dd::Edge edge, std::unordered_map<dd::NodePtr, double> &probs) {
     auto it = probs.find(edge.p);
-    if(it != probs.end()) {
+    if (it != probs.end()) {
         return CN::mag2(edge.w) * it->second;
     }
     double sum;
-    if(dd->isTerminal(edge)) {
+    if (dd->isTerminal(edge)) {
         sum = 1.0;
     } else {
         sum = assign_probs(edge.p->e[0], probs) + assign_probs(edge.p->e[2], probs);
     }
 
-    probs.insert(std::pair<dd::NodePtr , double>(edge.p, sum));
+    probs.insert(std::pair<dd::NodePtr, double>(edge.p, sum));
 
     return CN::mag2(edge.w) * sum;
 }
+
+dd::CTkind Simulator::ReturnNoiseOperation(char i, double prob) const {
```

Comment:
this function is non-void and should return something in either case.

Reply:
The function now returns dd::none in case of an error.